deps/Makefile

```diff
@@ -1957,26 +2002,28 @@ LIBGIT2_OPTS += -DCMAKE_FIND_ROOT_PATH=/usr/$(XC_HOST) -DCMAKE_FIND_ROOT_PATH_MO
 endif
 endif
 
-$(LIBGIT2_SRC_DIR)/build/Makefile: $(LIBGIT2_SRC_DIR)/CMakeLists.txt
-	-cd $(LIBGIT2_SRC_DIR) && patch -p1 -f < ../libgit2_oldcmake.patch
-	-cd $(LIBGIT2_SRC_DIR) && patch -p1 -f < ../libgit2_oldssl.patch
-	mkdir -p $(LIBGIT2_SRC_DIR)/build
-	cd $(LIBGIT2_SRC_DIR)/build/ && \
-	$(CMAKE) .. $(LIBGIT2_OPTS)
+$(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/patched: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/CMakeLists.txt
+	cd $(dir $<) && patch -p1 -f < $(SRCDIR)/libgit2_oldcmake.patch
+	cd $(dir $<) && patch -p1 -f < $(SRCDIR)/libgit2_oldssl.patch
+	echo 1 > $@
+$(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile: $(SRCDIR)/srccache/$(LIBGIT2_SRC_DIR)/patched
+	mkdir -p $(dir $@)
+	cd $(dir $@) && \
+	$(CMAKE) $(dir $<) $(LIBGIT2_OPTS)
 	touch -c $@
-$(LIBGIT2_OBJ_SOURCE): $(LIBGIT2_SRC_DIR)/build/Makefile
-	$(MAKE) -C $(LIBGIT2_SRC_DIR)/build
+$(LIBGIT2_OBJ_SOURCE): $(BUILDDIR)/$(LIBGIT2_SRC_DIR)/Makefile
+	$(CMAKE) --build $(dir $<)
```

Comment:
we may need to look into that further than. this change was already done for llvm and was being done here to be consistent across all cmake usage in the Makefile. it is in support of the following comment: https://github.com/JuliaLang/julia/pull/12463/files#diff-3ba529ae517f6b57803af0502f52a40bR57